src/context/ColonyContext.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { createContext, useMemo, ReactNode } from 'react';
+import { gql, useQuery } from '@apollo/client';
+import { useParams } from 'react-router-dom';
+import { defineMessages } from 'react-intl';
+
+import { getFullColonyByName } from '~gql';
+import { Colony } from '~types';
+import LoadingTemplate from '~frame/LoadingTemplate';
+import NotFoundRoute from '~routes/NotFoundRoute';
+import { useCanInteractWithColony } from '~hooks';
+
+interface ColonyContextValue {
+  colony?: Colony;
+  loading: boolean;
+  canInteractWithColony: boolean;
+}
+
+const ColonyContext = createContext<ColonyContextValue>({
+  loading: false,
+  canInteractWithColony: false,
+});
+
+const displayName = 'ColonyContextProvider';
+
+const MSG = defineMessages({
+  loadingText: {
+    id: `${displayName}.loadingText`,
+    defaultMessage: 'Loading Colony',
+  },
+});
+
+export const ColonyContextProvider = ({
+  children,
+}: {
+  children: ReactNode;
+}) => {
+  const { colonyName } = useParams<{ colonyName: string }>();
+
+  const { data, loading, error } = useQuery(gql(getFullColonyByName), {
+    skip: !colonyName,
+    variables: {
+      name: colonyName ?? '',
+    },
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const [colony] = data?.getColonyByName?.items || [];
+
+  const canInteractWithColony = useCanInteractWithColony(colony);
+
+  const colonyContext = useMemo<ColonyContextValue>(
```

Comment:
a nitpick but I think useMemo is not really necessary here, you can just create an object when passing the value. Or am I missing something?

Reply:
Use of memo is correct here. Since this is passed as a prop to the Provider component, which itself is a top level component, if not memoized, will re-render like crazy. 

And not just the first child, but the whole tree, which in this case is more or less the whole app.

Reply:
didn't think about it - good to know!